docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,1031 @@
+    classDef switchStyle fill:brown,color:white,stroke:none,rx:10px,ry:10px,font-size:25px;
+    classDef termStyle font-family:monospace,fill:black,stroke:none,color:white;
+    subgraph node2
+        subgraph GR-node2
+            rtos-GR-node2["rtos-GR-node2 100.65.0.3/16 (0a:58:64:41:00:03)"]
+        end
+        subgraph layer2-switch_node2["layer2-switch"]
+            stor-GR-node1["stor-GR-node1
+            type: <b>remote</b><br>requested-tnl-key: <b>4</b>"]
+            stor-GR-node2["stor-GR-node2
+            type: <b>router</b><br>requested-tnl-key: <b>5</b>"]
+        end
+    end
+    subgraph node1["node1"]
+        subgraph GR-node1
+            rtotr-GR-node1["trtor-GR-node1
+            100.65.0.2/16 100.88.0.5/31 (0a:58:64:41:00:02)"]
+        end
+        subgraph transit_router_node1["transit_router "]
+            trtor-GR-node1["trtor-GR-node1 100.88.0.4/31"]
+            rtos-layer2-switch["rtos-layer2-switch 203.203.0.1 (0a:58:CB:CB:00:01)"]
+        end
+        subgraph layer2-switch_node1["layer2-switch"]
+            stor-transit_router["stor-transit_router
+            type: <b>router</b><br>requested-tnl-key: <b>4</b>"]
+            stor-GR-node1-node2["stor-GR-node2
+            type: <b>remote</b><br>requested-tnl-key: <b>5</b>"]
+        end
+    end
+
+
+
+    rtotr-GR-node1 <--> trtor-GR-node1
+    rtos-layer2-switch <--> stor-transit_router
+    stor-GR-node1 <--> stor-transit_router
+    stor-GR-node2 <--> rtos-GR-node2
+    stor-GR-node1-node2 <--> stor-GR-node2
+
+    class VM vmStyle;
+    class rtotr-GR-node1 portStyle;
+    class rtos-GR-node2 portStyle;
+    class stor-GR-node2 portStyle;
+    class stor-GR-node1-node2 portStyle;
+    class stor-GR-node1 portStyle;
+    class trtor-GR-node1 portStyle;
+    class trtor-GR-node2 portStyle;
+    class stor-transit_router portStyle;
+    class rtos-layer2-switch portStyle;
+    class GR-node1 routerStyle;
+    class GR-node2 routerStyle;
+    class transit_router_node1 routerStyle;
+    class layer2-switch_node1 switchStyle;
+    class layer2-switch_node2 switchStyle;
+    class term termStyle;
+    class node1,node2 nodeStyle;
+```
+
+### Egress IP changes (TBD)
+
+### Transit subnet conflict
+
+Before this change Layer2 networks using Subnet overlapping with the `transitSubnet` were allowed to be created (and would be working just fine).
+Now it is not possible anymore, and we need to decide what to do in this case:
+1. Before upgrading to the new topology, check that no Layer2 networks with overlapping subnet are present.
+If they are, block the upgrade and inform the user to fix the issue. This option prevents the user from getting new topology
+for all networks until the next upgrade.
+2. Upgrade all nodes, but leave old topology for networks with overlapping subnet. Report an event/warning for the user to fix the issue.
+When the network is upgraded/re-created with non-overlapping subnet, it will get the new topology. Other networks will be
```

Comment:
how does one upgrade the networks without disruption to workloads if they were using 100.88 as the subnet?
to me it looks like we need to expose the transit subnet as a configurable field in UDN API right - similar to joinsubnets? how else can users configure the value to be non-overlapping because today its not configurable right?

Reply:
replied in another thread https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5485#discussion_r2329496114